### PR TITLE
style(patch): improve printin of the `pnpm patch` command

### PR DESCRIPTION
**Pull request by cpojer**
`pnpm patch` is great and I use it frequently. I love that it's integrated first-party into the package manager! This PR improves the output and usability of the command.

`pnpm patch` is different compared to most other pnpm commands in that it prints a folder and another command to run after patching the package. Usually those temporary folder names are long and the developer is likely to select the output and copy it. However, the output is a bit hard to read and the command has quotes around it, requiring the selection to be accurate in order not to copy-paste an invalid command.

This PR adds whitespace and color, and turns the folder into a link that can be clicked (cmd+click on MacOS).

### Before
CleanShot 2024-06-26 at 09 31 52@2x

### After
CleanShot 2024-06-26 at 09 45 34@2x

Text selection is much easier now with the whitespace around it since a user can now simply select the start of the text and drag the cursor down by two lines without accidentally copying other characters:
CleanShot 2024-06-26 at 09 45 50@2x

### CODE REVIEW

**Comment by zkochan** [comment]
Looks great. Please also add a changeset that describes your changes (as noted in CONTRIBUTING.md)

**Comment by zkochan** [comment]
also, looks like this library that you added is esm, so you cannot import it.

**Comment by cpojer** [comment]
Updated the changelog + downgraded to a version of `terminal-link` that doesn't use ESM (There have been no other changes to this package except making it ESM only, anyway).

**Review by zkochan — APPROVED** [review]

**Comment by zkochan** [comment]
The tests failed.

File: patching/plugin-commands-patching/package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "@pnpm/read-package-json": "workspace:*",
     "@pnpm/read-project-manifest": "workspace:*",
     "@pnpm/store-connection-manager": "workspace:*",
+    "chalk": "catalog:",
```
Comment (cpojer) [inline patching/plugin-commands-patching/package.json:61]: TIL `catalog:`!

**Comment by cpojer** [comment]
Thanks for fixing things up. Sorry, still getting used to the pnpm repo 🙇‍♂️

File: patching/plugin-commands-patching/package.json (hunk 1 above)
Comment (zkochan) [reply patching/plugin-commands-patching/package.json:61]: it was just released in beta

**Pull request merged**